.github/workflows/set-milestone-from-base-branch.yml

```diff
@@ -35,26 +35,28 @@ jobs:
           PR_NUMBER: ${{ github.event.number }}
         name: "Set milestone on related issues in brave-browser"
         run: |
-          URL_REGEX='(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)'
+          BB_ISSUE_SUFFIX="https://github.com/brave/brave-browser/issues/"
+          BB_ISSUE_URL_REGEX='(http(s)?:\/\/)?github.com/brave/brave-browser/issues/\d+'
           UPLIFT_PR_REGEX='uplift\s+(?:of)?:?\s+\K(#\d+)'
-          RESOLVE_ISSUE_REGEX="(?:resolv|fix|clos)(?:e|es|ed)?:?\s+\K(${URL_REGEX})"
+          RESOLVE_ISSUE_REGEX="(?:resolv|fix|clos)(?:e|es|ed)?:?\s+\K(${BB_ISSUE_URL_REGEX})"
+          verlte() { printf '%s\n%s' "$1" "$2" | sort -C -V; }
 
-          uplift_pr_body=$(gh pr view "$PR_NUMBER" -R "$REPO_NAME" --json body --jq '.body')
+          pr_body=$(gh pr view "${PR_NUMBER}" -R "${REPO_NAME}" --json body --jq .body | sed "s@\sbrave\/brave-browser\#\([0-9]\+\)@ ${BB_ISSUE_SUFFIX}\1@g")
 
-          issues=$(echo "${uplift_pr_body}" | grep -ioP "${RESOLVE_ISSUE_REGEX}" || :)
-          if [[ -z ${issues} ]]; then
-              echo "Resolved issues not directly found. Following PRs to determine them..."
-              prs=$(echo "${uplift_pr_body}" | grep -ioP "${UPLIFT_PR_REGEX}")
-              issues=$(xargs -n 1 -I{} gh pr view {} -R brave/brave-core --json body --jq '.body' < <(echo "${prs}") | grep -ioP "${RESOLVE_ISSUE_REGEX}")
-          fi
+          mapfile -t issues_pr < <(echo "${pr_body}" | grep -ioP "${RESOLVE_ISSUE_REGEX}" || :)
+          echo "Looking for any uplifted PRs"
+          mapfile -t prs < <(echo "${pr_body}" | grep -ioP "${UPLIFT_PR_REGEX}" || :)
+          mapfile -t issues_uplift_pr < <(echo "${prs[@]}" | xargs -n 1 -I{} gh pr view {} -R brave/brave-core --json body --jq ".body" | sed "s@\sbrave\/brave-browser\#\([0-9]\+\)@ ${BB_ISSUE_SUFFIX}\1@g" | grep -ioP "${RESOLVE_ISSUE_REGEX}")
+          declare -A issues
+          for i in "${issues_pr[@]}" "${issues_uplift_pr[@]}"; do issues["$i"]=1; done
```

Comment:
Here you're creating an associative array and setting keys to the values of the other two arrays and values to 1? That's pretty confusing/complicated.

The `issues` array is only used twice. Why not skip creating it altogether and just using the two arrays in those two places?

```bash
echo "Relevant issues detected:\n${issues_pr[*]} ${issues_uplift_pr[*]}"
```
and
```bash
for issue in "${issues_pr[@]}" "${issues_uplift_pr[@]}"; do
    # ...
done
```

Reply:
Done